src/Reflection/ReflectionParameter.php

```diff
@@ -126,7 +126,15 @@ public static function createFromNode(
         $param->isVariadic = (bool)$node->variadic;
         $param->isByReference = (bool)$node->byRef;
         $param->parameterIndex = (int)$parameterIndex;
-        $param->typeHint = (new FindTypeFromAst())->__invoke($node->type);
+        $param->typeHint = (new FindTypeFromAst())->__invoke(
+            $node->type,
+            $function->getLocatedSource(),
+            (
+                $function instanceof ReflectionMethod
```

Comment:
This is actually how TypeResolver works internally as well: https://github.com/phpDocumentor/TypeResolver/blob/master/src/Types/ContextFactory.php#L43